test/transactions/4_registerMultisignatureAccount.js

```diff
@@ -189,4 +202,77 @@ describe('#registerMultisignatureAccount transaction', () => {
 				.and.be.hexString();
 		});
 	});
+
+	describe('when the register multisignature account transaction is created with one too short public key', () => {
+		it('should throw an error', () => {
+			registerMultisignatureAccount
+				.bind(null, {
+					passphrase,
+					secondPassphrase,
+					keysgroup: tooShortPublicKeyKeysgroup,
+					lifetime,
+					minimum,
+				})
+				.should.throw(
+					'Public key d019a4b6fa37e8ebeb64766c7b239d962fb3b3f265b8d3083206097b912cd9 length differs from the expected 32 bytes for a public key.',
+				);
+		});
+	});
+
+	describe('when the register multisignature account transaction is created with one plus prepended public key', () => {
+		it('should throw an error', () => {
+			registerMultisignatureAccount
+				.bind(null, {
+					passphrase,
+					secondPassphrase,
+					keysgroup: plusPrependedPublicKeyKeysgroup,
+					lifetime,
+					minimum,
+				})
+				.should.throw('Invalid hex string');
+		});
+	});
+
+	describe('when the register multisignature account transaction is created with one empty keysgroup', () => {
+		it('should throw an error', () => {
+			registerMultisignatureAccount
+				.bind(null, {
+					passphrase,
+					secondPassphrase,
+					keysgroup: [],
+					lifetime,
+					minimum,
+				})
+				.should.throw(
+					'You must have between 1 and 16 public keys in the keysgroup.',
+				);
+		});
+	});
+
+	describe('when the register multisignature account transaction is created with 17 public keys in keysgroup', () => {
+		const generatePublicKey = () =>
+			cryptoModule.getPrivateAndPublicKeyFromPassphrase(
+				Math.random().toString(),
+			).publicKey;
```

Comment:
Prefer a deterministic test. E.g. calling this function on the index in the map function.